app/models/manageiq/providers/google/resource.rb

```diff
@@ -0,0 +1,40 @@
+module ManageIQ::Providers::Google
+  class Resource
+    def initialize(uri)
+      @type = self.class.parse_resource_type(uri)
+    end
+
+    attr_reader :type
+
+    def snapshot?
+      @type == 'snapshots'
+    end
+
+    def image?
+      @type == 'images'
+    end
+
+    def disk?
+      @type == 'disks'
+    end
+
+    # Parses a Google resource url and returns the resource from the path.
+    # This method understands both complete urls (starting with
+    # https://www.googleapis.com) as well as partials.
+    def self.parse_resource_type(uri)
+      return nil if uri.blank?
```

Comment:
how about returning `unknown` or `unsupported`? Returning `nil` is something one should avoid as much as possible